connectors/class-connector-editor.php

```diff
@@ -187,33 +186,34 @@ public function action_links( $links, $record ) {
 	}
 
 	/**
-	 * Retrieves data submitted on the screen, and prepares it for the appropriate context type
+	 * Retrieves data submitted on the screen, prepares it for the appropriate context type and logs the changes
 	 *
-	 * @action load-theme-editor.php
-	 * @action load-plugin-editor.php
+	 * @action wp_ajax_edit-theme-plugin-file
 	 */
 	public function get_edition_data() {
-		if (
-			(
-				isset( $_SERVER['REQUEST_METHOD'] )
-				&&
-				'POST' !== sanitize_text_field( $_SERVER['REQUEST_METHOD'] )
-			)
-			||
-			'update' !== wp_stream_filter_input( INPUT_POST, 'action' )
-		) {
+		$action         = wp_stream_filter_input( INPUT_POST, 'action' );
+		$request_method = wp_stream_filter_input( INPUT_SERVER, 'REQUEST_METHOD' );
+
```

Comment:
<!-- highlight specific problems -->
**❗️issue**
Can you add a nonce and capability check here? I think it can be the same capability check as is required to do the action.

Reply:
Fixed that as well.